Nautilus/Utility/ThunderkitUtilities/ApplySNMaterial.cs

```diff
@@ -0,0 +1,192 @@
+    private static Material _shinyGlassMaterial;
+    private static Material _interiorWindowGlassMaterial;
+    private static Material _holographicUIMaterial;
+    private static bool _cyclopsLoaded;
+
+    private void OnValidate()
+    {
+        if (!renderer) TryGetComponent(out renderer);
+    }
+
+    private IEnumerator Start()
+    {
+        yield return LoadMaterialsAsync();
+
+        if (!runAtStart) yield break;
+
+        yield return AssignMaterials();
+    }
+
+    /// <summary>
+    /// Applies the set material using the specified <see cref="MaterialApplicationMode"/>
+    /// A coroutine is required to ensure all material references are retrieved before being applied
+    /// </summary>
+    public IEnumerator AssignMaterials()
+    {
+        if (!_cyclopsLoaded)
+        {
+            yield return LoadMaterialsAsync();
+        }
+
+        switch(applicationMode)
+        {
+            case MaterialApplicationMode.SingleObject:
+                ApplyMaterialsOnSingleRend();
+                break;
+            case MaterialApplicationMode.AllChildObjects:
+                ApplyMaterialsOnChildren(false);
+                break;
+            case MaterialApplicationMode.AllChildObjectsIncludeInactive:
+                ApplyMaterialsOnChildren(true);
+                break;
+        }
+    }
+
+    private void ApplyMaterialsOnSingleRend()
+    {
+        if (renderer == null) throw new System.Exception($"The renderer is null on {gameObject} when SN materials were trying to be applied");
+
+        var mats = renderer.materials;
+        foreach (var index in materialIndices)
+        {
+            mats[index] = GetMaterial(materialType);
+        }
+
+        renderer.materials = mats;
+    }
+
+    private void ApplyMaterialsOnChildren(bool includeInactive)
+    {
+        var rends = GetComponentsInChildren<Renderer>(includeInactive);
+        foreach (var rend in rends)
+        {
+            var materials = rend.materials;
+            for (int i = 0; i < materials.Length; i++)
+            {
+                materials[i] = GetMaterial(materialType);
+            }
+
+            rend.materials = materials;
+        }
+    }
+
+    private Material GetMaterial(MaterialType type)
+    {
+        Material mat = type switch
+        {
+#if SN_STABLE
+            MaterialType.WaterBarrier => MaterialUtils.AirWaterBarrierMaterial,
+            MaterialType.ForceField => MaterialUtils.ForceFieldMaterial,
+            MaterialType.StasisField => MaterialUtils.StasisFieldMaterial,
+            MaterialType.HolographicUI => _holographicUIMaterial,
+#endif
+            MaterialType.Glass => _glassMaterial,
+            MaterialType.ExteriorGlass => _exteriorGlassMaterial,
+            MaterialType.ShinyGlass => _shinyGlassMaterial,
+            MaterialType.InteriorWindowGlass => _interiorWindowGlassMaterial,
+            _ => null
+        };
+
+        return mat;
+    }
+
+    private IEnumerator LoadMaterialsAsync()
```

Comment:
This method should be moved to `MaterialUtils` and called in `MaterialUtils.PatchInternal`